lib/text/listener.js

```diff
@@ -330,41 +328,53 @@ class Importer extends SHRParserListener {
     return vs;
   }
 
-  processFullyQualifiedCode(ctx) {
+  processFullyQualifiedCode(ctx, isValuesetDoc) {
     var cs;
     const csAlias = ctx.ALL_CAPS().getText();
-    const resolution = this._preprocessedData.resolveVocabulary(csAlias, this._currentNs, ...this._usesNs);
+    const resolution = this.getPreprocessedData(isValuesetDoc).resolveVocabulary(csAlias, this.getNamespace(isValuesetDoc), ...this._usesNs);
     if (resolution.error) {
       this.addError(resolution.error);
       cs = resolution.url ? resolution.url : csAlias;
     } else {
       cs = resolution.url;
     }
     const code = ctx.code().CODE().getText().substr(1); // substr to skip the '#'
-    const concept = new Concept(cs, code);
+    var concept;
+    if (isValuesetDoc) {
+      concept = new ValuesetConceptRule(cs, code);
+    } else {
+      concept = new Concept(cs, code);
+    }
+
     if (ctx.code().STRING()) {
       concept.display = stripDelimitersFromToken(ctx.code().STRING());
     }
     return concept;
   }
 
-    processInlineValue(ctx) {
-        var cs;
-        const csAlias = ctx.CODE().getText();
-        const resolution = this._preprocessedData.resolvePath('default', this._currentNs);
-        if (resolution.error) {
-            this.addError(resolution.error);
-            cs = resolution.url ? resolution.url : csAlias;
-        } else {
-            cs = resolution.url;
-        }
-        const code = ctx.CODE().getText().substr(1); // substr to skip the '#'
-        const concept = new Concept(cs, code);
-        if (ctx.STRING()) {
-            concept.display = stripDelimitersFromToken(ctx.STRING());
-        }
-        return concept;
-    }
+    processInlineValue(ctx, isValuesetDoc) {
+      var cs;
+
+      const csAlias = ctx.CODE().getText();
+      const resolution = this.getPreprocessedData(isValuesetDoc).resolvePath('default', this.getNamespace(isValuesetDoc));
```

Comment:
This returns the default path specified at the document level (e.g., `http://standardhealthrecord.org/foo/vs/`) -- but we really want something specific to the codes in this valueset.  It's not clear how this should be built, but here is what I suggest:
1. If `/vs/` exists in the `resolution.url`, replace it with `cs` (since we're talking now about a code system, not a value set).
2. If `/vs/` doesn't exist in the `resolution.url`, then append `cs/` to the end (or `/cs/` if the original url doesn't already end in `/`).
3. Finally, in either case above, append the valueset name at the end, just to scope the codes.

So, for example, consider the following:
```
ValueSetDefinitions: shr.foo 
Path: http://standardhealthrecord.org/foo/vs/

ValueSet: Stuff
#abc
#def
```
The codes (`abc` and `def`) should probably have the code system: `http://standardhealthrecord.org/foo/cs/Stuff`.